### ISSUE TITLE
ty: Fails to accept special forms as types

### ISSUE THREAD

**Issue by JelleZijlstra** [astral-sh/ruff#17905]
### Summary

```python
from collections.abc import Callable
from typing import Protocol

def gimme_ty(t: type):
    pass

gimme_ty(Callable)
gimme_ty(Protocol)
```

```
    Argument to this function is incorrect: Expected `type`, found `typing.Callable` (lint:invalid-argument-type) [Ln 7, Col 10]
    Argument to this function is incorrect: Expected `type`, found `typing.Protocol` (lint:invalid-argument-type) [Ln 8, Col 10]
```

https://play.ty.dev/3aebb756-4e06-414b-9445-b53a4b1b9f5f

Both of these are in fact types (instances of `builtins.type`) at runtime.

**Comment by my1e5** [comment]
I think this is the same error I'm seeing with `dataclasses` and `fields`?

https://play.ty.dev/331a08e9-3adf-41f4-a4a5-483a964a1b64

```py
from dataclasses import dataclass, fields


@dataclass
class Foo:
    x: int
    y: float
    z: str


for field in fields(Foo):
    print(field.name, field.type)
```

```
$ uvx ty check foo.py 
error: lint:invalid-argument-type: Argument to this function is incorrect
  --> foo.py:11:21
   |
11 | for field in fields(Foo):
   |                     ^^^ Expected `DataclassInstance`, found `Literal[Foo]`
12 |     print(field.name, field.type)
   |
info: Function defined here
   --> stdlib\dataclasses.pyi:220:5
    |
218 |     ) -> Any: ...
219 |
220 | def fields(class_or_instance: DataclassInstance | type[DataclassInstance]) -> tuple[Field[Any], ...]: ...
    |     ^^^^^^ -------------------------------------------------------------- Parameter declared here
221 |
222 | # HACK: `obj: Never` typing matches if object argument is using `Any` type.
    |
```

**Comment by JelleZijlstra** [comment]
That seems like a different issue to me.
